proposals/0001-hip-hyperfile/README.md

```diff
@@ -0,0 +1,160 @@
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This proposal introduces the concept of a `hyperfile.toml` configuration file for specifying
+sandbox settings in Hyperlight. This feature will allow users to define configurations such
+as `stack_size_override`, `max_execution_time`, and execution modes (`RunInHypervisor` or
+`RunInProcess`) in a structured and human-readable format. The goal is to enable host 
+applications built with Hyperlight to be configurable in a non-programmatic way, 
+allowing users to modify microVM settings without directly altering the host code.
+
+## Motivation
+
+### Goals
+
+- Enable the creation and use of a `hyperfile.toml` configuration file for defining sandbox 
+settings.
+- Support all current configuration options (e.g., `stack_size_override`, `max_execution_time`) 
+in the new format.
+- Maintain backwards compatibility with direct code-based sandbox configuration.
+
+### Non-Goals
+
+- This proposal does not aim to change how sandbox configurations are applied programmatically.
+- It does not address dynamic configuration updates while a sandbox is running.
+
+## Proposal
+
+The `hyperfile.toml` will serve as a configuration file for sandbox initialization. Users will 
+specify parameters in the TOML format. The Hyperlight API will expose methods to parse the 
+file and use it to initialize sandboxes.
+
+### User Stories
+
+#### Story 1: Simplifying Development
+
+As a developer, I want to define my sandbox settings in a `hyperfile.toml` so I can avoid 
+repetitive code when creating multiple sandboxes with similar configurations.
+
+#### Story 2: Easier Debugging and Sharing
+
+As a systems engineer, I want a standardized configuration file to share with teammates for
+debugging or reproducing issues without requiring them to modify source code.
+
+#### Story 3: Supporting Off-the-Shelf Hosts
+
+As a host provider offering an off-the-shelf host leveraging Hyperlight, I want to use a 
+`hyperfile.toml` to define configurable sandbox settings. This allows end users to adjust 
+microVM parameters without modifying the host's code, ensuring the integrity of the host 
+application while providing flexibility to users.
+
+### Notes/Constraints/Caveats
+
+- **Validation**: The parser must validate the TOML file to ensure all required fields are 
+provided and within acceptable ranges. Defaults will be used for any omitted optional fields.
+- **Error Handling**: Errors during parsing should provide detailed feedback to help users 
+correct their configuration files.
+
+### Risks and Mitigations
+
+#### Increased Complexity for New Users
+
+Introducing a configuration file may add initial complexity for users unfamiliar with TOML. 
+
+- **Mitigation**: Provide a well-documented template and examples in the Hyperlight 
+documentation.
+
+## Design Details
+
+### Rough Sketch of a `hyperfile.toml`
```

Comment:
Have you considered being able to specify multiple different named sandbox configurations in the same file that can be referenced by name when creating the sandboxes?

I can see a use case for wanting to specify different configurations.

This would work similar to how runtime classes work in containerd.

Reply:
If we do this I imagine one of the entries would be a default one which would be used if no named sandbox configuration is used.

Reply:
What about just doing this in separate files? `config1.toml` and `config2.toml`. It would simplify parsing alot, especially if we use the same struct for the config value itself, AND serde toml deserialization. And then you could do something like Config::FromFile(...) on the one you want to use.